xof.go

```diff
@@ -0,0 +1,44 @@
+package kyber
+
+import (
+	"crypto/cipher"
+	"io"
+)
+
+// An XOF is an extendable output function, which is a cryptographic
+// primitive that can take arbitrary input like a hash, and then create
+// a stream of output, up to a limit determined by the size of the internal
+// state of the XOF.
+type XOF interface {
+	// Write absorbs more data into the hash's state. It panics if called
+	// after Read. Use Reseed() return an XOF into a state where more data
+	// can be absorbed via Write.
+	io.Writer
+
+	// Read reads more output from the hash. It returns io.EOF if the limit
+	// has been reached.
+	io.Reader
+
+	// An XOF implements cipher.Stream, so that callers can use XORKeyStream
+	// to encrypt/decrypt data. The key stream is read from the XOF using
+	// the io.Reader interface.
```

Comment:
What happens if the limit is reached during XORKeyStream?

Reply:
Panic. But the limit for blake is 2^31 and there is no limit for sha3.Shake256.

Reply:
Should this be put in the comments?

Reply:
Done